crates/rome_js_formatter/src/utils/mod.rs

```diff
@@ -523,3 +524,51 @@ impl Format<JsFormatContext> for FormatMemberName {
         }
     }
 }
+
+/// This function is in charge to format the call arguments.
+pub(crate) fn fmt_arguments_multi_line<S: Format<JsFormatContext>, I>(
+    separated: I,
+    number_of_elements: usize,
+    f: &mut JsFormatter,
+) -> FormatResult<()>
+where
+    I: Iterator<Item = S>,
+    S: std::fmt::Debug,
+{
```

Comment:
Nit: You can require that `I implements ExactSizeIterator` so that you don't need to pass the length as well
```suggestion
pub(crate) fn fmt_arguments_multi_line<S: Format<JsFormatContext>, I>(
    separated: I,
    f: &mut JsFormatter,
) -> FormatResult<()>
where
    I: Iterator<Item = S> + ExactSizeIterator,
    S: std::fmt::Debug,
{
  let number_of_elements = separated.len();
```

Reply:
Or you can do 

```rust
let mut iterator = separated.enumerate().peekable();

while let Some((index, element)) = iterator.next() {
  let last = iterator.peek().is_none();

  if !last {
    join_with.entry(....)
  } else {
    ...
  }
} 
```